sympy/polys/polytools.py

```diff
@@ -4431,6 +4431,21 @@ def degree(f, gen=0):
 
 
 @public
+def total_degree(f, *gen):
+    f = sympify(f)
+    if f.is_Poly:
+        p = f.args[0]
```

Comment:
Do you mean gens instead of gen?
```
def total_degree(f, *gens):
  p = sympify(f)
  if p.is_Poly:
    p = p.as_expr()
  if p.is_Number:
    rv = 0
  else:
    if f.is_Poly:
      gens = gens or f.gens
    rv = Poly(p, *gens).total_degree()  
  return Integer(rv)

>>> total_degree(x+x*y+x*y*z, x)
1
>>> total_degree(x+x*y+x*y*z, x, y)
2
>>> total_degree(x+x*y+x*y*z, x, y, z)
3
>>> total_degree(x+x*y+x*y*z, x, z)
2
>>> total_degree(x+x*y+x*y*z, z)
1
>>> total_degree(x+x*y+x*y*z, y, z)
2
```

Reply:
> Do you mean gens instead of gen?

I meant `gen` but it should have been `gens`, since it takes more than one terms.

Also, @smichr what would be the answer of:

```
f = Poly(x**9 + x*z*y + x**3*z**2 + z**7,x)
total_degree(f,z))
```
I think it should be `9`. The above code would give `7`.

Reply:
I think @smichr 's code is doing the right thing. What it is doing is, using the generators specified as argument to `total_degree`, and in your example it is `z` as generator denying the generator `x` given by `Poly` itself.

You can see how that behaves in `degree` function:
```python
In [1]: f = Poly(x**9 + x*z*y + x**3*z**2 + z**7,x)

In [2]: f
Out[2]: Poly(x**9 + z**2*x**3 + y*z*x + z**7, x, domain='ZZ[y,z]')

In [3]: degree(f, y)
Out[3]: 1

In [4]: degree(f, z)
Out[4]: 7
```

Reply:
@gxyd but why the generator specified in `Poly` is denied. Shouldn't that be also included in `gens`.

```
According to above logic.

f = x**9 + x*z*y + x**3*z**2 + z**7
total_degree(f,x,z)

#this would return 9

But if I make it as:

f = Poly(x**9 + x*z*y + x**3*z**2 + z**7,x)
total_degree(f,z)
 
#this gives `7`.
```
But I guess if I am not wrong both means the same. So both should return the same result.

Reply:
I think this a choice of design, as to how things should work. And probably there is a reason for it. Now consider the design choice you suggest. If `f = Poly(x**9 + x*z*y + x**3*z**2 + z**7, x)` then by 'your' design choice can you find the degree w.r.t. (just) `z` of the polynomial `f`? No, you can't. So the 'current'(`degree's`) design choice is better since it is more flexible. That is how I see it. May be smichr has some more good point to add to it.

Reply:
Any factor that is not a generator is treated like a fancy number without order:

```
>>> degree(y**2, x)
0
```

So such factors are not part of the total degree (which gives an answer wrt generators).